smart_card_connector_app/src/window-apps-displaying.js

```diff
@@ -0,0 +1,121 @@
+/** @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This script implements displaying a list of connected
+ * applications in the Smart Card Connector App window.
+ */
+
+goog.provide('GoogleSmartCard.ConnectorApp.Window.AppsDisplaying');
+
+goog.require('GoogleSmartCard.DebugDump');
+goog.require('GoogleSmartCard.Logging');
+goog.require('GoogleSmartCard.PcscLiteServerClientsManagement.PermissionsChecking.KnownApp');
+goog.require('GoogleSmartCard.PcscLiteServerClientsManagement.PermissionsChecking.KnownAppsRegistry');
+goog.require('goog.array');
+goog.require('goog.asserts');
+goog.require('goog.dom');
+goog.require('goog.log.Logger');
+
+goog.scope(function() {
+
+/** @const */
+var GSC = GoogleSmartCard;
+
+/** @const */
+var PermissionsChecking = GSC.PcscLiteServerClientsManagement.PermissionsChecking;
+
+/** @const */
+var KnownApp = PermissionsChecking.KnownApp;
+
+/**
+ * @type {!goog.log.Logger}
+ * @const
+ */
+var logger = GSC.Logging.getScopedLogger('ConnectorApp.MainWindow');
+
+/**
+ * @type {Element}
+ * @const
+ */
+var appListElement = goog.dom.getElement('app-list');
+
+/**
+ * @const
+ */
+var knownAppsRegistry = new PermissionsChecking.KnownAppsRegistry();
+
+/**
+ * @type {goog.Promise.<!Array.<KnownApp>>}
+ */
+var lastKnownAppsPromise = null;
+
+/**
+ * @param {!goog.Promise.<!Array.<KnownApp>>} knownAppsPromise
+ * @param {!Array.<string>} appIds
+ * @param {Array.<KnownApp>} knownApps
+ * @return {boolean}
+ */
+var updateAppView = function(knownAppsPromise, appIds, knownApps) {
```

Comment:
So, what's the plan with the function definition syntax?
The proposed style will become a precedent in the existing project's codebase (with probably hundreds of already existing global functions), and the style is not presented in the Google style guide and almost (or completely?) not used in Closure Library source code.
And, by the way, looks like Closure Compiler anyway may transform this construct into function declaration syntax, effectively defeating the reasoning behind the function expression syntax: [snippet link](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540formatting%2520pretty_print%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250Ahello('New%2520user')%253B%250Ahello('New%2520user')%253B%250Ahello('New%2520user')%253B%250Avar%2520hello%2520%253D%2520function(name)%2520%257B%250A%2520%2520alert('Hello%252C%2520'%2520%252B%2520name)%253B%250A%257D%253B%250A).